main.go

```diff
@@ -127,7 +128,8 @@ func main() { //nolint:funlen,maintidx
 	ctrl.SetLogger(logger.ConfigLoggers(logmode))
 
 	// root context
-	ctx := ctrl.SetupSignalHandler()
+	rootCtx := ctrl.SetupSignalHandler()
+	ctx := logf.IntoContext(rootCtx, setupLog)
```

Comment:
not sure i understand why for this change: 
so the old code is, using rootCtx everywhere
the new code is, pass the new ctx which is `context.WithValue(rootCtx, contextKey{}, setupLog) ` instead except to start manager.

Reply:
the new  code should pass the original context without setuplog since I found it a bit incorrect to pass log.WithName("setup") further. Did I do a mistake?

Reply:
so at the beginning, we do `setupLog = ctrl.Log.WithName("setup")`
then later e.g 
```
if err = (&dscictrl.DSCInitializationReconciler{
		Client:                mgr.GetClient(),
		Scheme:                mgr.GetScheme(),
		Log:                   logger.LogWithLevel(ctrl.Log.WithName(operatorName).WithName("controllers").WithName("DSCInitialization"), logmode),
		Recorder:              mgr.GetEventRecorderFor("dscinitialization-controller"),
		ApplicationsNamespace: dscApplicationsNamespace,
	}).SetupWithManager(ctx, mgr); err != nil {
		setupLog.Error(err, "unable to create controller", "controller", "DSCInitiatlization")
		os.Exit(1)
	}
```
call with setupLog when problem to start contoller.
context is not used for for logging `"unable to create controller", "controller", "DSCInitiatlization".`
or?

Reply:
you mean ctx in .SetupWithManager(ctx, mgr)? It is not used at all IIUC, I planned patch to remove it.

Reply:
yeah, but what is "the new code should pass the original context without setuplog"? 

because from this change, except the `mgr.Start(rootCtx`) is kept to use the root context (regardless what the name of variable is now), everywhere else is using the new one (logf.IntoContext(rootCtx, setupLog)). so i think the way e.g  `upgrade.CreateDefaultDSCI(ctx,...) ` is starting to have "setup" with this PR vs it did not before the PR? which is  "the new code should pass the original context with setuplog" :thinking:

Reply:
Yes, that's the point. When (if) we switch to logger in context, everything before mgr.Start() is setup and will have "setup" logger, `upgrade.CreateDefaultDSCI()` as well. But `Reconcile` is not setup anymore. Of course, it can just ignore the logger from ctx, but I wanted to make it explicit.

Reply:
Ignore my last comment :) Let me think

Reply:
So `upgrade.CreateDefaultDSCI(ctx,...)` as executed by manager after `Start()` will have `ctx` without "setup" logger. Basically, what it uses now getting logger directly from the controller.

Reply:
Actually, controller framework passes its own logger with the context, so it's not needed to save rootCtx.

Reply:
@zdtsw I reverted that change. Any more concerns?